web/src/features/datasets/server/dataset-router.ts

```diff
@@ -570,6 +569,27 @@ export const datasetRouter = createTRPCRouter({
         skip: input.page * input.limit,
       });
 
+      const traceScores = await ctx.prisma.score.findMany({
+        where: {
+          projectId: ctx.session.projectId,
+          traceId: {
+            in: runItems
+              .filter((ri) => !!!ri.observationId) // only include trace scores if run is not linked to an observation
+              .map((ri) => ri.traceId),
+          },
+        },
+      });
+      const observationScores = await ctx.prisma.score.findMany({
+        where: {
+          projectId: ctx.session.projectId,
+          observationId: {
+            in: runItems
+              .filter((ri) => !!ri.observationId)
```

Comment:
double exclamation mark expected?

Reply:
see above, switched to !== null to be more explicit